ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/block/BlockProcessorAltair.java

```diff
@@ -176,10 +176,16 @@ public void processSyncCommittee(
     final MutableBeaconStateAltair state = MutableBeaconStateAltair.required(baseState);
     final SszVector<SszPublicKey> committeePubkeys = state.getCurrentSyncCommittee().getPubkeys();
     final List<BLSPublicKey> participantPubkeys = new ArrayList<>();
-    aggregate
-        .getSyncCommitteeBits()
-        .streamAllSetBits()
-        .forEach(index -> participantPubkeys.add(committeePubkeys.get(index).getBLSPublicKey()));
+    final List<BLSPublicKey> idlePubkeys = new ArrayList<>();
+
+    for (int i = 0; i < committeePubkeys.size(); i++) {
+      if (aggregate.getSyncCommitteeBits().getBit(i)) {
+        participantPubkeys.add(committeePubkeys.get(i).getBLSPublicKey());
```

Comment:
(nit) We could extract `committeePubkeys.get(i).getBLSPublicKey()` into a variable for better readability

Reply:
sure, this is just how the code was, but we're duplicating it i guess.